packages/astro/src/assets/fonts/vite-plugin-fonts.ts

```diff
@@ -250,14 +261,28 @@ export function fontsPlugin({ settings, sync, logger }: Options): Plugin {
 			if (isBuild) {
 				await initialize({
 					resolveMod: (id) => import(id),
+					base: new URL(CACHE_DIR, settings.config.cacheDir),
 				});
 			}
 		},
 		async configureServer(server) {
+			const localFontsWatcher = new LocalFontsWatcher();
+			localFontsWatcher.update = () => {
+				logger.info('assets', 'Font file updated');
+				server.restart();
+			};
 			const moduleLoader = createViteLoader(server);
 			await initialize({
 				resolveMod: (id) => moduleLoader.import(id),
+				// In dev, we cache fonts data in .astro so it can be easily inspected and cleared
+				base: new URL(CACHE_DIR, settings.dotAstroDir),
 			});
+			if (hashToUrlMap) {
+				localFontsWatcher.getPaths = () =>
+					[...hashToUrlMap!.values()].filter((url) => isAbsolute(url));
+			}
+			server.watcher.on('change', (path) => localFontsWatcher.onUpdate(path));
+			server.watcher.on('unlink', (path) => localFontsWatcher.onUnlink(path));
```

Comment:
Shouldn't we purge fonts when they are deleted too?

Reply:
Added a comment about that